actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes, variants: request.variant)
```

Comment:
I tried to follow the preceding comments, but I couldn't find why this isn't inside `default_render`. That's public API, and I think it's fair to consider the `:no_content` fallback to be part of the default rendering behaviour.